code/modules/unit_tests/geyser.dm

```diff
@@ -0,0 +1,27 @@
+///Geysers listen to reagent signals to know when to start processing, which is very cool, smart, optimized and fragile
+/datum/unit_test/geyser/Run()
+	//While we're at it just check em all
+	var/list/geysers = subtypesof(/obj/structure/geyser)
+	for(var/geyser_type as anything in geysers)
+		var/obj/structure/geyser/wittel/geyser = allocate(geyser_type)
+		geyser.potency = geyser.max_volume //make it recharge in 1 tick
+
+		TEST_ASSERT(geyser.reagents, "Geyser does not have a reagent datum! Source: [geyser.type]")
+		TEST_ASSERT(geyser.reagents.has_reagent(geyser.reagent_id), "Geyser should start with [geyser.reagent_id], but started with [geyser.reagents.get_reagent_log_string()] instead. Source: [geyser.type]")
+
+		geyser.reagents.clear_reagents() //this should awaken the geyser to start refilling
+
+		var/do_we_do_autorefill = FALSE
+		var/time_passed = 0 SECONDS
+		//shit can be delayed due to laggines or something, so check every 2 seconds. if it lasts longer than 30 somethings definitely fucking wrong
+		while(time_passed < 30 SECONDS)
```

Comment:
1. You have to change the priority of the test, or you hold up smaller tests which makes running tests locally more annoying
2. You already shouldn't really be sleeping in tests, but 30 seconds is definitely too high IMO. I'm not thrilled with how long our tests are but the ones that are long catch very common regressions, this is definitely past my limit. What are we doing here?